src/main/scala/rise/core/types/Type.scala

```diff
@@ -3,7 +3,47 @@ package rise.core.types
 import arithexpr.arithmetic.RangeAdd
 import rise.core._
 
-sealed trait Type
+object alphaEquiv {
+  val equiv : Type => Type => Boolean = a => b => {
+    (a.getClass == b.getClass && b.getClass == TypePlaceholder.getClass) ||
+    (a.getClass == b.getClass && b.getClass == NatType.getClass) ||
```

Comment:
why not add these two cases in the `match` below?

Reply:
This is a leftover from when `Type.equals` called `alphaEquiv.equiv`, as matching against objects calls `.equals` behind the scenes, so we were getting into infinite loops. I still need to rewrite alpha equivalence on terms so that it doesn't use substitution, like I did with types (potentially much bigger speed up? fingers crossed), I will do it as part of that.